variable/pow.cpp

```diff
@@ -51,12 +51,13 @@ template <class T> struct PowUnit {
 template <class V>
 Variable pow_handle_unit(V &&base, const Variable &exponent,
                          const bool in_place) {
-  if (exponent.unit() != units::one) {
+  if (const auto exp_unit = variableFactory().elem_unit(exponent);
+      exp_unit != units::one) {
```

Comment:
Maybe just throw if exponent is binned? Or is there a working application of this?

Reply:
I don't know if this is needed. I think in principle, there is no reason not to support it. But it seems like a rare use case.
It was not supported anyway because of https://github.com/scipp/scipp/blob/e586fafe327fd6fa91cda3cfcdfec3b11c3f2f7c/variable/pow.cpp#L75-L76
So I just added an explicit check for now.